app/crud/geostore.py

```diff
@@ -115,3 +132,253 @@ async def create_user_area(geometry: Geometry) -> Geostore:
+        )
+
+        where_clauses.append(
+            db.text(f"gid_{adm_level} LIKE :level_id_pattern").bindparams(
+                level_id_pattern=level_id_pattern
+            )
+        )
+
+    sql: Select = db.select(columns_etc).select_from(src_table)
+
+    for clause in where_clauses:
+        sql = sql.where(clause)
+
+    row = await get_first_row(sql)
+    if row is None:
+        raise RecordNotFoundError(
+            f"Admin boundary not found in {admin_provider} version {admin_version}"
+        )
+
+    if row.geojson is None:
+        raise GeometryIsNullError(
+            "GeoJSON is None, try reducing or eliminating simplification."
+        )
+
+    geostore: AdminGeostore = await form_admin_geostore(
+        adm_level=adm_level,
+        admin_version=admin_version,
+        area=float(row.gfw_area__ha),
+        bbox=[float(val) for val in row.gfw_bbox],
+        name=str(row.name),
+        geojson=json.loads(row.geojson),
+        geostore_id=str(row.gfw_geostore_id),
+        level_id=str(row.level_id),
+        simplify=simplify,
+    )
+
+    return AdminGeostoreResponse(data=geostore)
+
+
+async def admin_params_to_dataset_version(
+    source_provider: str, source_version: str
+) -> Tuple[str, str]:
+    admin_source_to_dataset: Dict[str, str] = {"GADM": "gadm_administrative_boundaries"}
+
+    try:
+        dataset: str = admin_source_to_dataset[source_provider.upper()]
+    except KeyError:
+        raise BadAdminSourceException(
+            (
+                "Invalid admin boundary source. Valid sources:"
+                f" {[source.lower() for source in admin_source_to_dataset.keys()]}"
+            )
+        )
+
+    try:
+        version: str = per_env_admin_boundary_versions[ENV][source_provider.upper()][
+            source_version
+        ]
+    except KeyError:
+        raise BadAdminVersionException(
+            (
+                "Invalid admin boundary version. Valid versions:"
+                f" {[v for v in per_env_admin_boundary_versions[ENV][source_provider.upper()].keys()]}"
+            )
+        )
+
+    return dataset, version
+
+
+async def form_admin_geostore(
+    adm_level: int,
+    bbox: List[float],
+    area: float,
+    geostore_id: str,
+    level_id: str,
+    simplify: Optional[float],
+    admin_version: str,
+    geojson: Dict,
+    name: str,
+) -> AdminGeostore:
+    info = Adm0BoundaryInfo.parse_obj(
+        {
+            "use": {},
+            "simplifyThresh": simplify,
+            "gadm": admin_version,
+            "name": name,
+            "iso": extract_level_id(0, level_id),
+        }
+    )
+    if adm_level >= 1:
+        info = Adm1BoundaryInfo(
+            **info.dict(),
+            id1=int(extract_level_id(1, level_id)),
+        )
+    if adm_level == 2:
+        info = Adm2BoundaryInfo(
+            **info.dict(),
+            id2=int(extract_level_id(2, level_id)),
+        )
+
+    return AdminGeostore.parse_obj(
```

Comment:
Also can be done later, but I feel like it'd be more elegant to have AdminGeostore have a constructor to just take these params, and then have a function that outputs them in this format. But maybe just me prefernce.

Reply:
Hmm. I think I prefer it this way because it's how we tend to do it elsewhere and I'd like to be consistent. But that is an interesting idea.